tests/unit/checkpoint/test_mics_optimizer.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import deepspeed
+
+from unit.common import DistributedTest
+from unit.simple_model import *
+
+from unit.checkpoint.common import *
+
+
+class TestMiCSCheckpoint(DistributedTest):
+    world_size = 4
+
+    def _toy_model_config(self):
+
+        config_dict = {
+            "train_micro_batch_size_per_gpu": 2,
+            "steps_per_print": 1,
+            "optimizer": {
+                "type": 'Adam',
+                "params": {
+                    "lr": 0.00015,
+                    "betas": [0.8, 0.999],
+                    "eps": 1e-8,
+                    "weight_decay": 3e-7
+                }
+            },
+            "fp16": {
+                "enabled": True,
+                "initial_scale_power": 8
+            },
+            "wall_clock_breakdown": True,
+            "zero_optimization": {
+                "stage": 3,
+                "mics_shard_size": 2
```

Comment:
Is it possible to test different `mics_shard_size` values, e.g., 1, 2, and 4?

Reply:
Yes, it is possible for 2 and 4. Not sure the effect on 1, because if 1 is used, it means zero-2.

Reply:
I guess you mean 1 would be vanilla zero-3. The expectation is that it would be functional but not performant, right? I think that is okay for a unit test as it will help us catch unexpected breakages.

Reply:
my mistake. For `"mics_shard_size": 1`, I think it is equivalent to DDP, in which each device holds a complete copy of the model states (grads, params, optimizer states)

Reply:
Thanks for the clarification. I think it would still be a good test case.

Reply:
I see, thanks for the suggestions, I added more test as you suggested

Reply:
Sorry, I don't see the modification to make shard size configurable in the unit test. Perhaps, I am missing something. 

What I was expecting was something like
```python
@pytest.mark.parametrize('shard_size', [1, 2, 4])
class TestMiCSCheckpoint(DistributedTest):
   def _toy_model_config(self, shard_size):
    ...
     "mics_shard_size": shard_size
    ...
    def test_load_optimizer_state(self, tmpdir, shard_size):   
      ....
      config_dict, hidden_dim, models = self._toy_model_config(shard_size)

```

Reply:
sry, I was looking into a wrong file. 
The tests are updated with the shard_size [1,2,4] covered.

Reply:
Awesome! Thanks!